language/move-prover/src/bytecode_translator.rs

```diff
@@ -1360,17 +1360,17 @@ impl<'env> ModuleTranslator<'env> {
                             src,
                             resource_type,
                         );
-                        emitln!(self.writer, &update_and_track_local(dest, "$tmp"));
+                        emitln!(self.writer, &propagate_abort());
```

Comment:
Hmm... actually we were doing the type check already right before, but on dest and not on $tmp (dest is the right place). Do you know whether this actually fixes the issue? It should be the same as before.

The mode should be still WellFormed::Default, which includes the invariant.

Reply:
Yes, it was doing it on `str_local(dest).as_str()`, which returns something like `$t3`. So, I modified it to do it with `"$tmp"`.

Reply:
$tmp is assigned to whatever dst is right away. So something is wrong with this fix.

Reply:
I suggest you comment out the failing ensures and undo this specific fix, then I take a deeper look after your PR landed. This fix is not right as it is, and we need to understand why.

Reply:
My update made the generated boogie code look like:
```
// Call(AttrId(1), [4], MoveFrom(ModuleId(0), StructId(Symbol(37)), []), [3])
call $tmp := $MoveFrom($GetLocal($m, $frame + 3), $TestResources_R_type_value());
if ($abort_flag) {
  assume $DebugTrackAbort(2, 1408);
  goto Abort;
}
assume $TestResources_R_is_well_formed_types($tmp);
$m := $UpdateLocal($m, $frame + 4, $tmp);
```
Since you think it needs a further investigation, I will undo my change on this.

Reply:
It's definitely a bug, but the bug is more universal than the fix you did. The same problem is likely elsewhere too. That's why I ask to hold back on this fix. Let me know when the PR is ready for review again.

Reply:
I undid my fix, and commented out the `ensures` statements in `resource.move` that triggers the bug.